test/e2e/service_test.go

```diff
@@ -17,93 +17,93 @@ package e2e
-		require.NoError(t, err, "Pod %s should be able to connect %s, but was not able to connect", podName, net.JoinHostPort(svc.Spec.ClusterIP, fmt.Sprint(80)))
-	}
+	nginx := fmt.Sprintf("nginx-%v", isIPv6)
+	hostNginx := fmt.Sprintf("nginx-host-%v", isIPv6)
+	clusterIP := createClusterIPService(t, data, isIPv6)
+	url := net.JoinHostPort(clusterIP, "80")
 
-	t.Run("ClusterIP", func(t *testing.T) {
-		t.Run("Same Linux Node can access the Service", func(t *testing.T) {
-			t.Parallel()
-			testFromPod("hostnetwork-client-on-same-node", serverPodNode, true)
-		})
-		t.Run("Different Linux Node can access the Service", func(t *testing.T) {
-			t.Parallel()
-			skipIfNumNodesLessThan(t, 2)
-			testFromPod("hostnetwork-client-on-different-node", nodeName(1), true)
-		})
-		t.Run("Windows host can access the Service", func(t *testing.T) {
-			t.Parallel()
-			skipIfNoWindowsNodes(t)
-			idx := clusterInfo.windowsNodes[0]
-			winNode := clusterInfo.nodes[idx].name
-			testFromNode(winNode)
-		})
-		t.Run("Linux Pod on same Node can access the Service", func(t *testing.T) {
-			t.Parallel()
-			testFromPod("client-on-same-node", serverPodNode, false)
-		})
-		t.Run("Linux Pod on different Node can access the Service", func(t *testing.T) {
-			t.Parallel()
-			skipIfNumNodesLessThan(t, 2)
-			testFromPod("client-on-different-node", nodeName(1), false)
-		})
+	createAndWaitForPod(t, data, data.createNginxPodOnNode, nginx, nodeName(0), testNamespace, false)
+	time.Sleep(2 * time.Second)
+	t.Run("Pod CIDR Endpoints", func(t *testing.T) {
+		testClusterIPCases(t, data, url, nodes, busyboxes)
 	})
+	require.NoError(t, data.deletePod(testNamespace, nginx))
+
+	createAndWaitForPod(t, data, data.createNginxPodOnNode, hostNginx, nodeName(0), testNamespace, true)
+	time.Sleep(2 * time.Second)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		testClusterIPCases(t, data, url, nodes, busyboxes)
+	})
+	require.NoError(t, data.deletePod(testNamespace, hostNginx))
 }
 
-func (data *TestData) createClusterIPServiceAndBackendPods(t *testing.T, name string, node string) (*corev1.Service, func()) {
-	ipv4Protocol := corev1.IPv4Protocol
-	require.NoError(t, data.createNginxPodOnNode(name, testNamespace, node))
-	_, err := data.podWaitForIPs(defaultTimeout, name, testNamespace)
-	require.NoError(t, err)
-	require.NoError(t, data.podWaitForRunning(defaultTimeout, name, testNamespace))
-	svc, err := data.createNginxClusterIPService(name, false, &ipv4Protocol)
-	require.NoError(t, err)
+func testClusterIPCases(t *testing.T, data *TestData, url string, nodes, pods []string) {
+	t.Run("All Nodes can access Service ClusterIP", func(t *testing.T) {
+		skipIfProxyAllDisabled(t, data)
+		skipIfKubeProxyEnabled(t, data)
+		for _, node := range nodes {
+			testClusterIPFromNode(t, url, node)
+		}
+	})
+	t.Run("Pods from all Nodes can access Service ClusterIP", func(t *testing.T) {
+		t.Parallel()
```

Comment:
why only this test runs in parallel? This seems pointless when just one test is doing this

Reply:
Removed.